File: app/src/main/java/net/treelzebub/zinepress/zine/Zine.java
package net.treelzebub.zinepress.zine;

import org.joda.time.LocalDate;

import java.io.Serializable;
import java.util.List;

import nl.siegmann.epublib.domain.Resource;

/**
 * Created by Tre Murillo on 1/8/16
 */
public class Zine implements Serializable {

    private Long              date;
    private String            title;
    private Resource          coverImage;
    private List<ZineArticle> zineArticles;

    public Zine(Long date, String title, Resource coverImage, List<ZineArticle> zineArticles) {
        this.date         = date;
        this.title        = title;
        this.coverImage   = coverImage;
        this.zineArticles = zineArticles;
    }

    public Long getDate() {
        return date;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Resource getCoverImage() {
        return coverImage;
    }

    public void setCoverImage(Resource coverImage) {
        this.coverImage = coverImage;
    }

    public List<ZineArticle> getZineArticles() {
        return zineArticles;
    }

    public void setZineArticles(List<ZineArticle> zineArticles) {
        this.zineArticles = zineArticles;
    }
}

File: app/src/main/java/net/treelzebub/zinepress/zine/Zine.kt
package net.treelzebub.zinepress.zine

import net.treelzebub.zinepress.db.zines.IZine
import net.treelzebub.zinepress.util.extensions.bytes
import java.util.*

/**
 * Created by Tre Murillo on 1/8/16
 */
class Zine : IZine {

    override val id: Long
    override val date: Long
    override val title: String
    override val articles: ByteArray

    constructor(id: Long, date: Long, title: String, zineArticles: HashSet<ZineArticle>) {
        this.id           = id
        this.date         = date
        this.title        = title
        this.articles     = zineArticles.bytes()
    }
}
